Add accessibility tests for admin Dashboard and Enrollments

PaulMorris opened the pull request:
Add accessibility tests for the admin Dashboard and Enrollments tabs (Draft, Pending, etc.).  Includes some refactoring to make existing test code reusable for these tests.

The changes in this PR are on top of those in the current version of PR #676.  No need to review until #676 lands on master and this is then rebased on master.  (The new commits start with "Refactor integration tests 'login' code for reusability".)

Tested by running the tests. Those that are not ignored (`@ignore`) pass successfully, and the ones that are ignored would fail with an accessibility issue.

Issue #518 Implement accessibility checking in CI

jasonaowen commented:
Jenkins, retest this please.

jasonaowen commented:
@PaulMorris it looks like [the dashboard page has an accessibility issue on Jenkins](https://ci.psm.solutionguidance.com/job/integration-tests/69/Browser=chromium/testReport/(root)/General%20Accessibility%20Checks%20for%20Admins/Admin_Dashboard_Page/).

Locally, I have several accessibility issues relating to #672.

PaulMorris commented:
@jasonaowen Hmm, I assume this is because I don't have the right enrollments in my database to cause these failures, so they are working locally for me.  All of these pages would be affected by that, so I can `@ignore` them when I rebase on master.

jasonaowen commented:
@PaulMorris do you think it might be an issue with the accessibility tests running on an empty database before the test that creates an enrollment?

PaulMorris commented:
@jasonaowen Yeah, good call!  That's probably it.  The admin a11y tests are being run first since 'admin' comes before 'enrollment' alphabetically.

PaulMorris commented:
I emptied my database and ran the tests and there are about 7 admin a11y tests that fail because they expect data to be in the database.  The easiest fix for this particular problem would be to rename the "admin" feature directory so that it comes after the "enrollment" and/or "general" directories, or vice-versa.  (I assume we don't want to add testing empty database states to the accessibility tests at this point.)

cecilia-donnelly commented:
@PaulMorris Agreed, we don't need to test empty database states right now.

PaulMorris commented:
More details: 

Most of the a11y test failures are caused by lack of data (enrollments) in the database, and can be fixed by renaming the "features/admin" directory so these tests are run after other tests that populate the database.  I think calling it "features/service_admin" (to go along with "system_admin") will work nicely for that.

But there's more.  Currently none of the tests actually submit a "Head Start" enrollment, and so some of the a11y tests for viewing organization provider enrollments still won't pass (because there is no "Head Start" enrollment to be found under the admin enrollments "Pending" tab).  To fix this we should have a test "submit" a "Head Start" enrollment.

Finally, even after running all the tests, populating an empty database, the admin dashboard page still has red error text under "Notfications" ("No matched data found.").  So that a11y test will still fail until #685 lands to fix the red text issue.

PaulMorris commented:
This revision addresses the issues discussed above.

It renames the `features/admin` directory to `features/service_admin` (for both the cucumber feature files and the java steps files) so those tests run after the enrollment tests, after the database is populated, if it was empty before.  

Also rebased on master with PR #685 in place, to solve the red text issue on the admin dashboard page.

In a revision of the next PR #684 I added a test that submits a "Head Start" enrollment, so that we have one of those in the database so we can access and test certain pages in the admin part of the site.

PaulMorris commented:
Rebased on latest version of PR #676.

PaulMorris commented:
This revision `@ignore`s the a11y tests (admin dashboard and enrollments pages) that can fail due to issue #672 when there are enrollments with no NPI in the database.
